packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -822,21 +831,48 @@ export class MergeTree {
 		if (!segment || !isRemovedAndAcked(segment)) {
 			return segment;
 		}
+
+		if (this.cachedSlideDestination?.seq !== segment.removedSeq) {
+			this.cachedSlideDestination = {
+				segmentToSlideDestination: new Map<ISegment, ISegment>(),
+				seq: segment.removedSeq,
```

Comment:
i don't think this assert is really necessary. i think typesciprt will already know that this.cachedSlideDestination can't be undefined, due to the above !isRemovedAndAcked(segment) which focrces removedSeq to not be undefined, so the above condition will only occur if  this.cachedSlideDestination is undefined.